src/swupd_lib/statedir.c

```diff
@@ -56,44 +66,128 @@ char *statedir_get_tracking_file(const char *bundle_name)
 	return sys_path_join("%s/%s/%s", globals.state_dir, TRACKING_DIR, bundle_name);
 }
 
+char *statedir_get_telemetry_record(char *record)
+{
+	return sys_path_join("%s/%s/%s", globals.state_dir, TELEMETRY_DIR, record);
+}
+
+char *statedir_get_swupd_lock(void)
+{
+	return sys_path_join("%s/%s", globals.state_dir, LOCK);
+}
+
+char *statedir_get_version(void)
+{
+	return sys_path_join("%s/%s", globals.state_dir, VERSION_FILE);
+}
+
+/* ************* */
+/* Cache section */
+/* ************* */
+
+static char *get_url(void)
+{
+	size_t i, j = 0;
+	char *url = strdup_or_die(globals.content_url);
+	char *converted_url = malloc_or_die(str_len(url) + 1);
+
+	// replace ':' and '/' with '_', but do not allow
+	// multiple '_' in a row
```

Comment:
I personally wouldn't avoiding multiple '_'. There's no problem on having that, reduces the chance on having url clashes and the code is easier.

Reply:
oo, the only bad thing is that makes the name extra long (which already is)